include/deal.II/grid/grid_tools.h

```diff
@@ -2089,6 +2102,128 @@ namespace GridTools
             }
         }
   }
+
+
+
+  template <typename Iterator>
+  Point<Iterator::AccessorType::space_dimension>
+  project_to_object(const Iterator &object,
+                    const Point<Iterator::AccessorType::space_dimension> &trial_point)
+  {
+    const unsigned int spacedim = Iterator::AccessorType::space_dimension;
+    const unsigned int structdim = Iterator::AccessorType::structure_dimension;
+
+    if (structdim == 1 && spacedim <= 1)
+      return trial_point;
+    else if (structdim == 2 && spacedim <= 2)
+      return trial_point;
+    else if (structdim == 3 && spacedim <= 3)
+      return trial_point;
```

Comment:
Since you have to have `structdim <= spacedim`, these three conditions all boil down to `if (structdim == spacedim)`.